test/functional/apps/saved_objects_management/edit_saved_object.ts

```diff
@@ -99,5 +121,48 @@ export default function({ getPageObjects, getService }: FtrProviderContext) {
       const objects = await PageObjects.settings.getSavedObjectsInTable();
       expect(objects.includes('A Dashboard')).to.be(false);
     });
+
+    it('preserves the object references when saving', async () => {
```

Comment:
testing that references are properly displayed and preserved during `save`, then testing that updating the references actually works.